acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/ValidatorKeysAcceptanceTest.java

```diff
@@ -0,0 +1,154 @@
+    final String networkName = "less-swift";
+    final BesuNode eth1Node = createBesuNode();
+    eth1Node.start();
+
+    final ValidatorKeystores validatorKeystores =
+        createTekuDepositSender(networkName).sendValidatorDeposits(eth1Node, 8);
+    final ValidatorKeystores extraKeys =
+        createTekuDepositSender(networkName).sendValidatorDeposits(eth1Node, 1);
+
+    final TekuNode beaconNode =
+        createTekuNode(config -> config.withNetwork(networkName).withDepositsFrom(eth1Node));
+
+    final TekuValidatorNode validatorClient =
+        createValidatorNode(
+                config ->
+                    config
+                        .withNetwork(networkName)
+                        .withValidatorApiEnabled()
+                        .withInteropModeDisabled()
+                        .withBeaconNode(beaconNode))
+            .withMutableValidatorKeystores(validatorKeystores);
+
+    beaconNode.start();
+    validatorClient.start();
+
+    getValidatorListing(validatorClient, 0);
+
+    addValidatorsAndExpect(validatorClient, validatorKeystores, tempDir, "imported");
+
+    validatorClient.waitForLogMessageContaining("Added validator");
+    validatorClient.waitForLogMessageContaining("Published block");
+    getValidatorListing(validatorClient, 8);
+
+    // second add attempt would be duplicates
+    addValidatorsAndExpect(validatorClient, validatorKeystores, tempDir, "duplicate");
+
+    // a random key won't be found, remove should give not_found
+    removeValidatorAndCheckStatus(validatorClient, extraKeys.getPublicKeys().get(0), "not_found");
+
+    // wait for epoch 1 to ensure the validator has attested, which will give slashing protection
+    // data
+    beaconNode.waitForLogMessageContaining("Epoch Event *** Epoch: 2");
+    // remove a validator
+    final BLSPublicKey removedPubkey = validatorKeystores.getPublicKeys().get(0);
+    removeValidatorAndCheckStatus(validatorClient, removedPubkey, "deleted");
+
+    // should only be 7 validators left
+    validatorClient.waitForLogMessageContaining("Removed validator");
+    validatorClient.waitForLogMessageContaining("Published block");
+    getValidatorListing(validatorClient, 7);
+
+    // remove the same validator again
+    removeValidatorAndCheckStatus(validatorClient, removedPubkey, "not_active");
+
+    validatorClient.stop();
+
+    beaconNode.stop();
+    eth1Node.stop();
+  }
+
+  private void addValidatorsAndExpect(
+      final TekuValidatorNode validatorClient,
+      final ValidatorKeystores validatorKeystores,
+      final Path tempDir,
+      final String expectedStatus)
+      throws IOException {
+    final JsonNode addResult =
+        jsonProvider
+            .getObjectMapper()
+            .readTree(validatorClient.addValidators(validatorKeystores, tempDir));
+    assertThat(addResult.get("data").size()).isEqualTo(validatorKeystores.getValidatorCount());
+    checkStatus(addResult.get("data"), expectedStatus);
+  }
```

Comment:
Can we push this into the DSL rather than doing it in the test?  So we'd have `validatorClient.addValidators(validatorKeystores, expectedStatus)` (it can make its own temp dir if it needs one).

That way if any other ATs need to add validators in the future they don't have to duplicate this code, and we keep the AT itself to be working at a fairly high level.

Same with the other private methods here - they'd be better pushed into the `TekuValidatorNode` class.

It might be worth putting them in a new class so that you'd have `validatorClient.getApi().addValidator`.  Maybe even just use a `public final` variable so you have`validatorClient.api.addValidator(..)`. It's a bit unusual but keeps the syntax really succinct in the AT which is the key aim here.

Reply:
its pretty specific to this flow, i was just refactoring out code, but i'll see if i can make it somewhat re-usable...